### PR TITLE
Check srcAccessMask in transitions from PRESENT_SRC_KHR

### PR DESCRIPTION
**Pull request by Tony-LunarG**
From the vkAcquireNextImageKHR section of the spec:

The vkCmdWaitEvents or vkCmdPipelineBarrier used to transition the image away from VK_IMAGE_LAYOUT_PRESENT_SRC_KHR layout must have dstStageMask and dstAccessMask parameters set based on the next use of the image. The srcAccessMask must include VK_ACCESS_MEMORY_READ_BIT to ensure that all prior reads by the presentation engine are complete before the image layout transition occurs.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -10701,6 +10701,45 @@ TEST_F(VkLayerTest, InvalidBarriers) {
...
+    range.baseArrayLayer = 0;
+    range.layerCount = 1;
+    barrier.subresourceRange = range;
+    VkCommandBufferObj cmdbuf(m_device, m_commandPool);
+    cmdbuf.BeginCommandBuffer();
+    cmdbuf.PipelineBarrier(VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, 0, 0, nullptr, 0, nullptr, 1,
+                           &barrier);
+    barrier.oldLayout = VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
+    barrier.newLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+    cmdbuf.PipelineBarrier(VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, 0, 0, nullptr, 0, nullptr, 1,
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: Error is generated at this point. No need to actually submit the command buffer.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -8723,6 +8723,15 @@ static bool ValidateBarriers(const char *funcName, VkCommandBuffer cmdBuffer, ui
                                          dst_q_f_index, dev_data->phys_dev_properties.queue_family_properties.size());
                 }
             }
+            if ((mem_barrier->oldLayout == VK_IMAGE_LAYOUT_PRESENT_SRC_KHR) &&
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: This should really be done in `ValidateMaskBitsFromLayouts`, like all the other similar constraints.

**Comment by Tony-LunarG** [comment]
Fixed

**Comment by chrisforbes** [comment]
@Tony-LunarG what do you want to do about that edge case you raised in email, where the layout doesn't change?

**Comment by Tony-LunarG** [comment]
I wasn't sure enough about the optimization to do it.  Are the src and dst access masks meaningless if the layout isn't changing?  If so, I'll add the optimization to this request

**Comment by Tony-LunarG** [comment]
Cherry picked to master

**Pull request closed without merging**